daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ExpressionEvaluatingParsers.scala

```diff
@@ -143,7 +143,9 @@ class AssertExpressionEvaluationParser(
         start.setDiscriminator(discrim)
       } else {
         val message = getAssertFailureMessage(start)
-        val diag = new AssertionFailed(decl.schemaFileLocation, start, message)
+        val currentElem = start.infoset
+        val details = "\nParsed value was: " + currentElem.toString
```

Comment:
Can this toString possibly fail? 
I think there should be Assert.invariant checks here which check explicitly the assumptions you are making that reassure you that this toString cannot fail, i.e., that a currentElem exists, that if it is of simple type, then it positively does have a value always by this point.

Reply:
Just pushed up a change that should hopefully address these concerns.